src/parameterized_tendencies/microphysics/precipitation.jl

```diff
@@ -325,30 +325,30 @@ function compute_precipitation_surface_fluxes!(
     (; surface_rain_flux, surface_snow_flux) = p.precipitation
     (; col_integrated_precip_energy_tendency,) = p.conservation_check
     (; ᶜwᵣ, ᶜwₛ, ᶜspecific) = p.precomputed
-
-    (; ᶠtemp_scalar) = p.scratch
-    slg = Fields.level(Fields.local_geometry_field(ᶠtemp_scalar), Fields.half)
-
-    # Constant extrapolation: - put values from bottom cell center to bottom cell face
-    ˢρ = Fields.Field(Fields.field_values(Fields.level(Y.c.ρ, 1)), axes(slg))
-    # For density this is equivalent with ᶠwinterp(ᶜJ, Y.c.ρ) and therefore
-    # consistent with the way we do vertical advection
-    ˢqᵣ = Fields.Field(
+    ᶜlg = Fields.local_geometry_field(Y.c)
+    sfc_lg = Fields.level(Fields.local_geometry_field(Y.f), Fields.half)
+    sfc_space = axes(sfc_lg)
+
+    # Jacobian-weighted extrapolation from interior center to surface face,
+    # consistent with the 3D quantity ᶠρ = ᶠinterp(Y.c.ρ * ᶜJ) / ᶠJ
+    int_lg = Fields.Field(Fields.field_values(Fields.level(ᶜlg, 1)), sfc_space)
+    int_ρ = Fields.Field(Fields.field_values(Fields.level(Y.c.ρ, 1)), sfc_space)
+    sfc_ρ = @. lazy(int_ρ * int_lg.J / sfc_lg.J)
+
+    # Constant extrapolation to surface, consistent with simple downwinding
+    sfc_qᵣ = Fields.Field(
         Fields.field_values(Fields.level(ᶜspecific.q_rai, 1)),
-        axes(slg),
+        sfc_space,
     )
-    ˢqₛ = Fields.Field(
+    sfc_qₛ = Fields.Field(
         Fields.field_values(Fields.level(ᶜspecific.q_sno, 1)),
-        axes(slg),
+        sfc_space,
     )
-    ˢwᵣ = Fields.Field(Fields.field_values(Fields.level(ᶜwᵣ, 1)), axes(slg))
-    ˢwₛ = Fields.Field(Fields.field_values(Fields.level(ᶜwₛ, 1)), axes(slg))
-
-    # Project the flux to CT3 vector and convert to physical units.
-    @. surface_rain_flux =
-        -projected_vector_data(CT3, ˢρ * ˢqᵣ * Geometry.WVector(ˢwᵣ), slg)
-    @. surface_snow_flux =
-        -projected_vector_data(CT3, ˢρ * ˢqₛ * Geometry.WVector(ˢwₛ), slg)
+    sfc_wᵣ = Fields.Field(Fields.field_values(Fields.level(ᶜwᵣ, 1)), sfc_space)
+    sfc_wₛ = Fields.Field(Fields.field_values(Fields.level(ᶜwₛ, 1)), sfc_space)
+
+    @. surface_rain_flux = sfc_ρ * sfc_qᵣ * (-sfc_wᵣ)
+    @. surface_snow_flux = sfc_ρ * sfc_qₛ * (-sfc_wₛ)
```

Comment:
@trontrytel I've also simplified the vector shenanigans I previously recommended for this.